library/common/extensions/filters/http/platform_bridge/filter.h

```diff
@@ -14,17 +17,51 @@ namespace Extensions {
 namespace HttpFilters {
 namespace PlatformBridge {
 
+/**
+ * All PlatformBridge Filter stats. @see stats_macros.h
+ */
+#define ALL_PLATFORM_BRIDGE_FILTER_STATS(COUNTER, HISTOGRAM)                                       \
+  HISTOGRAM(init_callback_latency, Milliseconds)                                                   \
+  HISTOGRAM(on_rq_headers_callback_latency, Milliseconds)                                          \
+  HISTOGRAM(on_rq_data_callback_latency, Milliseconds)                                             \
+  HISTOGRAM(on_rq_trailers_callback_latency, Milliseconds)                                         \
+  HISTOGRAM(on_rq_resume_callback_latency, Milliseconds)                                           \
+  HISTOGRAM(on_rs_headers_callback_latency, Milliseconds)                                          \
+  HISTOGRAM(on_rs_data_callback_latency, Milliseconds)                                             \
+  HISTOGRAM(on_rs_trailers_callback_latency, Milliseconds)                                         \
+  HISTOGRAM(on_rs_resume_callback_latency, Milliseconds)                                           \
+  HISTOGRAM(on_cancel_callback_latency, Milliseconds)                                              \
+  HISTOGRAM(on_error_callback_latency, Milliseconds)
+
+/**
+ * Struct definition for all DNS Filter stats. @see stats_macros.h
+ */
+struct PlatformBridgeFilterStats {
+  ALL_PLATFORM_BRIDGE_FILTER_STATS(GENERATE_COUNTER_STRUCT, GENERATE_HISTOGRAM_STRUCT)
+};
+
 class PlatformBridgeFilter;
 
 class PlatformBridgeFilterConfig {
 public:
   PlatformBridgeFilterConfig(
+      Server::Configuration::FactoryContext& context,
       const envoymobile::extensions::filters::http::platform_bridge::PlatformBridge& proto_config);
 
+  PlatformBridgeFilterStats& stats() const { return stats_; }
   const std::string& filter_name() { return filter_name_; }
   const envoy_http_filter* platform_filter() const { return platform_filter_; }
 
 private:
+  static PlatformBridgeFilterStats generateStats(const std::string& stat_prefix,
```

Comment:
Why have the stat prefix if we only set it to the empty string?

Reply:
Ah, good question, it's
1) because I Considered making it the filter name and think that still might be worth considering, but I held off because I'm not sure how consistently/well those are formatted for use in stats,

and
2) because this is basically the signature used everywhere in Envoy for this pattern.